platform/ios/PolyPodApp/OIDAuth/OIDAuth.swift

```diff
@@ -0,0 +1,137 @@
+            }
+        // TBD end session request
+    }
+    
+    /// Loads the Auth state.
+    /// - Parameter completion: The result of the load operation.
+    ///                         After a successful load, the `state` peroperty will be initialized.
+    func loadAuthState(completion: @escaping (Result<Void, Error>) -> Void) {
+        if let state = try? OIDTokenStorage.getAuthState(forService: authorizationRequest.clientID).get() {
+            self.state = state
+            completion(.success(()))
+            return
+        }
+        
+        self.currentAuthFlow = OIDAuthState.authState(
+            byPresenting: authorizationRequest,
+            externalUserAgent: oidExternalUserAgent) { state, error in
+            completion(Result {
+                switch (state, error) {
+                case let (state?, nil):
+                    self.state = state
+                case let (nil, error?):
+                    throw error
+                default:
+                    throw InvalidAuthResponseRepresentation()
+                }
+            })
+        }
+    }
+    
+    /// Handles Auth redirect to complete the Auth flow.
+    /// After redirect was handled, any subsequent calls to this function will be ignored,
+    /// unless a new auth flow is started.
+    /// - Parameter url: The auth redirect URL
+    func handleAuthRedirect(_ url: URL) {
+        currentAuthFlow?.resumeExternalUserAgentFlow(with: url)
+        currentAuthFlow = nil
+    }
+}
+
+extension OIDAuth: OIDAuthStateChangeDelegate {
+    // Handle state changes when token is refreshed or invalidated.
+    func didChange(_ state: OIDAuthState) {
+        self.state = state
+    }
+}
+
+extension OIDAuth {
+    /// An user agent which does not trigger any actual authorization flow.
+    /// Meant to be used when the authorization flow happens on the backend side,
+    /// while the app "just" needs to handle the code/token exchange.
+    /// This is necessary to "trick" AppAuth that a full OAuth2.0 flow is happening form the client side.
+    private final class OIDExternalUserAgentNoOp: NSObject, OIDExternalUserAgent {
+        func present(_ request: OIDExternalUserAgentRequest, session: OIDExternalUserAgentSession) -> Bool {
+            return true
+        }
+        
+        func dismiss(animated: Bool, completion: @escaping () -> Void) {
+            completion()
+        }
+    }
+    
+    /// Membership feature Auth configuration
+    static func membershipFeatureAuth() -> OIDAuth {
+        let redirectURL = URL(string: "coop.polypoly.polypod://membership_feature/oauth")!
+        let serviceConfig = OIDServiceConfiguration(
+            // No authorization endpoint, as auhorization it happens directly on backend side
+            authorizationEndpoint: URL(string: "")!,
+            tokenEndpoint: URL(
+                string: "https://keycloak.stage.polypoly.tech/realms/eu-members/protocol/openid-connect/token"
```

Comment:
Have you considered using an external variable here?

Reply:
This ideally should have been given by openid-configuration discovery, but we have settled that PolyPod should configure the token endpoints on its own. I would keep this here until we have a final decision on how we handle the OpenID endpoints to avoid making more complex changes.